cmd/git-init/main.go

```diff
@@ -66,10 +68,30 @@ func main() {
 		logger.Fatalf("Unexpected error creating symlink: %v", err)
 	}
 
+	dir, err := os.Getwd()
+	if err != nil {
+		logger.Errorf("Failed to get current dir", err)
+	}
+
+	if *name != "" {
+		// create dir name
+		path := filepath.Join(dir, *name)
+		if _, err := os.Stat(path); os.IsNotExist(err) {
```

Comment:
any reason for not using `os.MkdirAll` that would do the same almost ?

Reply:
I did not want to use `os.MkdirAll` because I expect the `workspace` dir to already exist and `name` path not to exist. I want to create name sub directory only if it doesn't exist. `os.MkdirAll` wont return an error if dir path exists which defeats the purpose.

Reply:
@shashwathi ah good point :sweat_smile:

Reply:
What if I want to check out the `k8s.io/test-infra` source to `/workspace/go/src/k8s.io/test-infra` rather than just `/workspace/test-infra`?

This is often an important use case for go programs.